lit_gpt/speed_monitor.py

```diff
@@ -239,10 +239,11 @@ def on_train_batch_end(
             # Assumes no padding.
             if lengths is not None:
                 elapsed_lengths = int(self.history_lengths[-1]) - int(self.history_lengths[0])
+                avg_length = elapsed_lengths / elapsed_batches
```

Comment:
Thanks for the reporting this issue. Why is it batches and not `elapsed_samples`?

Reply:
In all the scripts, we always only add the `input_ids.size(1)` not the total number of tokens per batch, so the normalization has to be per batch instead of per sample.